src/index.ts

```diff
@@ -10,12 +10,48 @@ import type { GeneratedFile, SourceFile } from '@kapeta/codegen';
 import { mergeDevcontainers } from './target/merge-devcontainers';
 import { addTemplateHelpers } from './target/template-helpers';
 import { mergePom } from './target/merge-pom';
+import {parseKapetaUri} from "@kapeta/nodejs-utils";
+import {BlockDefinition, BlockDefinitionSpec, Resource} from "@kapeta/schemas";
 
 export default class JavaSpringBootTarget extends Target {
     constructor(options: any) {
         super(options, Path.resolve(__dirname, '../'));
     }
 
+    generate(data: any, context: any): GeneratedFile[] {
+        let blockSpec = data as BlockDefinition;
+        const uri = parseKapetaUri('kapeta/resource-type-auth-jwt-provider');
+        const matcher = (consumer: Resource) => parseKapetaUri(consumer.kind).fullName === uri.fullName;
+        const hasJwtProvider = blockSpec?.spec?.providers?.some(matcher);
+
+        if (hasJwtProvider) {
+            let resource: Resource = {
```

Comment:
Not a huge fan of this - there are potential name collissions etc that make this less than ideal . 

I'd suggest "/.well-known/jwks.json" is something that is provided by our SDK and we then expect some sort of interface or config parameters - that might have a insecure default but can be overwritten with a more secure alternative. 

We use a simple file based keystore by default in the Node implementation: https://github.com/kapetacom/codegen-target-nodejs/blob/78033cc50ac01f0147345a19086229f02598cf52/templates/kapeta/resource-type-auth-jwt-provider/keystore-provider.ts.hbs#L5 
And allow the user to provide a more sophisticated one if they want

Reply:
In spring I'd think this would be even easier - since we can have a default bean provider of "JWKSStore" which the user can then overwrite by providing their own

Reply:
Changed this so codegen only enables configuration and the rest is in the sdk-spring-security.